TYPO3.Neos/Classes/TYPO3/Neos/Controller/Service/NodesController.php

```diff
@@ -98,7 +99,11 @@ public function indexAction($searchTerm = '', $workspaceName = 'live', array $di
         }
 
         $contentContext = $this->createContentContext($workspaceName, $dimensions);
-        $nodes = $this->nodeSearchService->findByProperties($searchTerm, $searchableNodeTypeNames, $contentContext, $contextNode);
+        if ($nodeIdentifiers === array()) {
+            $nodes = $this->nodeSearchService->findByProperties($searchTerm, $searchableNodeTypeNames, $contentContext, $contextNode);
+        } else {
+            $nodes = $this->nodeSearchService->findByIdentifiers($nodeIdentifiers, $contentContext, $contextNode);
```

Comment:
@dfeyer Using

    array_map(function($identifier) { return $contentContext->getNodeByIdentifier($identifier); }, $nodeIdentifiers)

would almost implement the new search service method and we don't need to refactor yet another method for the UI rewrite, so why implement a new search service method?

My issue with this is that every new code that uses the NodeDataRepository is a big smell and makes it harder to refactor later.